sched/clock/clock_systime_ticks.c

```diff
@@ -100,11 +103,15 @@ clock_t clock_systime_ticks(void)
        */
 
 #if defined(CONFIG_SCHED_TICKLESS_TICK_ARGUMENT)
-      clock_t ticks;
+      clock_t ticks = 0;
       up_timer_gettick(&ticks);
       return ticks;
 #elif defined(CONFIG_SCHED_TICKLESS)
-      struct timespec ts;
+      struct timespec ts =
+        {
+          0
+        };
+
       up_timer_gettime(&ts);
```

Comment:
maybe it's better to check the return value to initialize the clock?

Reply:
@XuNeo Please evaluate which implementation has the least impact on performance

![Screenshot from 2023-09-18 18-09-15](https://github.com/apache/nuttx/assets/758493/4fa6464e-e481-4219-bc6f-7c56583c4c31)

Reply:
Check return code is better than init `ticks` to zero by 3.9%.
```
/* Option 1 */
clock_t ticks = 0;
up_timer_gettick(&ticks);
return ticks;

/* Option 2 */
clock_t ticks;
if (up_timer_gettick(&ticks) == OK)
  return ticks;
else
  return 0;
```

Run a loop to call clock() for 1M times on cortex-M55 project, it costs 20593 ticks if init `ticks` while only 19796 ticks for check return code method.

Test code:
```
  start = clock();

  loop = 1000000;
  while(loop--)
    {
      clock();
    }

  end = clock();
```